modules/exporter-common/src/main/scala/io/janstenpickle/trace4cats/export/QueuedSpanCompleter.scala

```diff
@@ -27,30 +27,32 @@ object QueuedSpanCompleter {
   ): Resource[F, SpanCompleter[F]] = {
     val realBufferSize = if (bufferSize < batchSize * 5) batchSize * 5 else bufferSize
 
-    def write(inFlight: Ref[F, Int], queue: Queue[F, CompletedSpan], exporter: SpanExporter[F, Chunk]): F[Unit] =
-      queue.dequeue
-        .groupWithin(batchSize, batchTimeout)
-        .map(spans => Batch(spans))
-        .evalMap { batch =>
-          exporter.exportBatch(batch).guarantee(inFlight.update(_ - batch.spans.size))
-        }
-        .compile
-        .drain
-        .onError { case th =>
-          Logger[F].warn(th)("Failed to export spans")
-        }
-
     for {
       inFlight <- Resource.liftF(Ref.of(0))
       hasLoggedWarn <- Resource.liftF(Ref.of(false))
       queue <- Resource.liftF(Queue.bounded[F, CompletedSpan](realBufferSize))
-      _ <- Resource.make(
-        Stream
-          .retry(write(inFlight, queue, exporter), 5.seconds, _ + 1.second, Int.MaxValue)
+      _ <- Resource.make {
+        queue.dequeue
+          .groupWithin(batchSize, batchTimeout)
+          .map(spans => Batch(spans))
+          .evalMap { batch =>
+            Stream
+              .retry(
+                exporter.exportBatch(batch).onError { case th =>
+                  Logger[F].warn(th)("Failed to export spans")
+                },
```

Comment:
I think this `onError` would be better outside of the retry now.

Reply:
I don't think it's ever going to be called outside the retry because of the `maxAttempts = Int.MaxValue`?

Reply:
Yeah, good point. Thinking about it some more, I don't think this should repeat indefinitely, maybe 10 times at most (eventually making it configurable)